### PR TITLE
Don't crash when a path has invalid characters

### PR DESCRIPTION
**Pull request by paulvanbrenk**
Make sure we don't crash when a path has invalid characters.

### CODE REVIEW

File: Nodejs/Product/Nodejs/SharedProject/ProjectNode.cs [hunk 1]
```diff
@@ -6511,15 +6511,24 @@ public virtual int ParseCanonicalName(string name, out uint itemId)
...
+                var child = FindNodeByFullPath(name);
+                if (child != null)
+                {
+                    itemId = child.HierarchyId;
+                    return VSConstants.S_OK;
+                }
+            }
+            catch (ArgumentException)
+            {
+                // This is expected when the path contains
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/SharedProject/ProjectNode.cs]: Do we want to try to return a narrower HRESULT in this case?

Comment (amcasey) [inline Nodejs/Product/Nodejs/SharedProject/ProjectNode.cs]: If we're using Path utilities, it seems like IO exceptions might also be possible.

Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/SharedProject/ProjectNode.cs]: Doc's say ArgumentException is the only one, since the method doesn't do any actual IO.

Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/SharedProject/ProjectNode.cs]: The only path related return code was 'Path too long' which didn't seem appropriate.

**Pull request merged**